src/libraries/System.Private.Xml/src/System/Xml/Xsl/QIL/QilVisitor.cs

```diff
@@ -241,7 +243,7 @@ protected virtual QilNode Visit(QilNode n)
         protected virtual QilNode VisitReference(QilNode n)
         {
             if (n == null)
-                return VisitNull();
+                return VisitNull()!;
```

Comment:
IIRC some implementations didn't accept nulls. Null is handled only in some node types and some other do not need to handle it. Since for virtual methods you have to pick one type of annotation it makes code cleaner to leave this as non-nullable (I initially tried using NotNullIfNotNull but compiler expected all overrides to also have that).

Also note that all of these types are internal and it probably doesn't make sense to spend too much time on getting those perfect